webdriver/tests/bidi/script/add_preload_script/add_preload_script.py

```diff
@@ -235,3 +239,81 @@ async def add_preload_script_of_type(type):
         expected_result["value"].append({"type": "string", "value": type})
 
     assert result == expected_result
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("create_type", ["popup", "iframe"])
+@pytest.mark.parametrize("access_type", ["evaluate", "evaluate2", "opener", "opener2", "inline"])
+async def test_preload_script_properties_available_immediately(
+    bidi_session, add_preload_script, new_tab, subscribe_events, wait_for_event, wait_for_future_safe, create_type, access_type
+):
+    await add_preload_script(function_declaration="() => { window.foo = 'bar'; }")
+
+    await subscribe_events([CONTEXT_CREATED_EVENT, CONTEXT_LOAD_EVENT])
+    on_created = wait_for_event(CONTEXT_CREATED_EVENT)
+
+    if create_type == "popup":
+        if access_type == "evaluate":
+            script = "window.open('about:blank')"
+        elif access_type == "evaluate2":
+            script = "window.open()"
+        elif access_type == "opener":
+            script = "window.baz = window.open('about:blank').foo"
+        elif access_type == "opener2":
+            script = "window.baz = window.open().foo"
+        elif access_type == "inline":
+            script = "window.open('data:text/html,<script>window.baz = window.foo</script>')"
+    elif create_type == "iframe":
+        script = "const iframe = document.createElement('iframe');"
+        if access_type == "evaluate":
+            script += "iframe.src='about:blank'; document.body.appendChild(iframe)"
+        elif access_type == "evaluate2":
+            script += "document.body.appendChild(iframe)"
+        elif access_type == "opener":
+            script += """iframe.src='about:blank'; document.body.appendChild(iframe);
+                window.baz = iframe.contentWindow.foo"""
+        elif access_type == "opener2":
+            script += "document.body.appendChild(iframe); window.baz = iframe.contentWindow.foo"
+        elif access_type == "inline":
+            script += """iframe.src='data:text/html,<script>window.baz = window.foo</script>';
+                document.body.appendChild(iframe)"""
+
+    asyncio.create_task(
```

Comment:
Should we finally wait for the task to be completed before running the next test?

Reply:
I don't think that's necessary. We wait for the new context to be created, which implies that the evaluation that creates it completed.